integration-tests/pkg/executor/executor_k8s.go

```diff
@@ -0,0 +1,181 @@
+package executor
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	coreV1 "k8s.io/api/core/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	TESTS_NAMESPACE = "collector-tests"
+)
+
+type K8sExecutor struct {
+	clientset *kubernetes.Clientset
+}
+
+func newK8sExecutor() (*K8sExecutor, error) {
+	fmt.Println("Creating k8s configuration")
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		fmt.Printf("Error: Failed to get cluster config: %s\n", err)
+		return nil, err
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		fmt.Printf("Error: Failed to create client: %s", err)
+		return nil, err
+	}
+
+	k8s := &K8sExecutor{
+		clientset: clientset,
+	}
+	return k8s, nil
+}
+
+func (e *K8sExecutor) CopyFromHost(src string, dst string) (string, error) {
+	return "", fmt.Errorf("Unimplemented")
+}
+
+func (e *K8sExecutor) PullImage(image string) error {
+	return fmt.Errorf("Unimplemented")
+}
+
+func (e *K8sExecutor) IsContainerRunning(podName string) (bool, error) {
+	pod, err := e.clientset.CoreV1().Pods(TESTS_NAMESPACE).Get(context.Background(), podName, metaV1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+
+	if pod == nil || pod.Status.Phase != coreV1.PodRunning {
+		return false, nil
+	}
+
+	return pod.Status.ContainerStatuses[0].Ready, nil
+}
+
+type PodFilter struct {
+	Name      string
+	Namespace string
+}
+
+func (e *K8sExecutor) ContainerID(podFilter interface{}) string {
+	pf, ok := podFilter.(PodFilter)
+	if !ok {
+		return ""
+	}
+
+	pod, err := e.ClientSet().CoreV1().Pods(pf.Namespace).Get(context.Background(), pf.Name, metaV1.GetOptions{})
+	if err != nil {
+		fmt.Printf("%s\n", err)
+		return ""
+	}
+
+	if len(pod.Status.ContainerStatuses) != 1 {
+		return ""
+	}
+
+	containerID := pod.Status.ContainerStatuses[0].ContainerID
+	if len(containerID) < 12 {
+		fmt.Printf("Invalid container ID: %q\n", containerID)
+		return ""
+	}
+
+	i := strings.LastIndex(containerID, "/")
+	if i == -1 {
+		fmt.Printf("Invalid container ID: %q\n", containerID)
+		return ""
+	}
+
+	return containerID[i+1 : i+13]
```

Comment:
Because the container id looks something like this when you get it from k8s client:
```
containerd://1234567890abcdef...
```
`string.LastIndex` gets the position of the second `/` and we want 12 characters for the short container ID, I was too lazy to do `i++`, so instead the slice goes from +1 to +13, I can add the `i++` or a comment describing this if it helps.

Reply:
A little comment to describe why this is needed would be useful I think

Reply:
Replaced that final return statement to use `common.ContainerShortID` and added a small comment with an example of the format extracted from the k8s client SDK.